libcyphal_demo/src/platform/linux/can/can_media.hpp

```diff
@@ -0,0 +1,277 @@
+// This software is distributed under the terms of the MIT License.
+// Copyright (C) OpenCyphal Development Team  <opencyphal.org>
+// Copyright Amazon.com Inc. or its affiliates.
+// SPDX-License-Identifier: MIT
+// Author: Sergei Shirokov <sergei.shirokov@zubax.com>
+
+#ifndef PLATFORM_LINUX_CAN_MEDIA_HPP_INCLUDED
+#define PLATFORM_LINUX_CAN_MEDIA_HPP_INCLUDED
+
+#include "platform/posix/posix_executor_extension.hpp"
+#include "platform/posix/posix_platform_error.hpp"
+#include "socketcan.h"
+
+#include <canard.h>
+#include <cetl/pf17/cetlpf.hpp>
+#include <cetl/pf20/cetlpf.hpp>
+#include <cetl/rtti.hpp>
+#include <libcyphal/executor.hpp>
+#include <libcyphal/transport/can/media.hpp>
+#include <libcyphal/transport/errors.hpp>
+#include <libcyphal/types.hpp>
+
+#include <algorithm>
+#include <cerrno>
+#include <cstddef>
+#include <cstdint>
+#include <iterator>
+#include <string>
+#include <unistd.h>
+#include <utility>
+#include <vector>
```

Comment:
CAN platform files are not yet adopted for "no c++ heap" - they are currently copy/pasted from libcyphal examples as is.